### PR TITLE
Fix some mods being disallowed for playlist

### PR DESCRIPTION
**Pull request by nanaya**
I completely forgot about it when actually writing the code 👀

### CODE REVIEW

**Review by notbakaneko — COMMENTED** [review]
should this handle the `UserPlayable` `false` cases?

**Comment by nanaya** [comment]
Neither seem a breaking problem so maybe later. There's also configuration requirement which isn't checked at the moment either.

**Comment by notbakaneko** [comment]
As far as I can tell, `ValidForMultiplayer: false` always has an associated `UserPlayable: false`, so non-realtime cases that were legitimately failing before would be passing now?

**Comment by nanaya** [comment]
Anyone weird enough going through effort allowing cinema/autoplay mods probably already expect weird result in the first place.

**Comment by peppy** [comment]
Let's deploy this early next week in case there's some new missed cases.

File: app/Libraries/Mods.php [hunk 1]
```diff
@@ -88,6 +88,17 @@ public function __construct()
         }
     }
 
+    public function checkAllModsMeta(int $rulesetId, array $mods, callable $check, callable $failItemCallback): void
```
Comment (notbakaneko) [inline app/Libraries/Mods.php:91]: Everything in this function is external to the class other than the call to a `public $this->mods`, it looks like it should go in `PlaylistItem`

**Review by notbakaneko — CHANGES_REQUESTED** [review]

**Pull request merged**